File: 2211-count-collisions-on-a-road/2211-count-collisions-on-a-road.java
class Solution {
    public int countCollisions(String dir) {
        
        
        	int left = 0 , right = dir.length()-1;

	while (left < dir.length() && dir.charAt(left) =='L'){
		left ++; 
	}
	while (right >=0 && dir.charAt(right) =='R'){
		right --;
	}

	int cnt =0;
	for (int i=left ; i <=right; i++){
		if (dir.charAt(i) != 'S') {
			cnt ++;
		}
	}
	return cnt; 
    }
}

File: 2211-count-collisions-on-a-road/2211-count-collisions-on-a-road.py
class Solution:
    def countCollisions(self, dir: str) -> int:



        right =0 ; res = 0 ; obstacle = False 

        for d in dir:

            if d=='R':
                right +=1 
            elif d=='S':
                res += right 
                right = 0 
                obstacle = True 

            else: # d=='L'
                if right > 0:
                    res += right +1 
                    right = 0 
                    obstacle = True 
                elif obstacle:
                    res +=1

        return res 